### DISCUSSION TITLE
Planar Antenna Gain

### DISCUSSION THREAD

**Discussion by nourhan0106** [NVlabs/sionna#554] [General]
Hi,

I am trying to configure an omnidirectional antenna with gain 4dBi. I am not sure how to scale the gain of the planar antenna in Sionna as it is normalized to 1. I hope you can guide me through this. 
Also is there a way to set the sensitivity and the gain of the receive antenna?

**Comment by fanyimeng0** [comment]
As far as I know, you can set the antenna gain in the antenna pattern. [ link](https://nvlabs.github.io/sionna/_modules/sionna/rt/antenna.html#iso_pattern). Similarly, you could also be able to set the sensitivity and gain of the receiving antenna, similar to the tr38901_pattern [link](https://nvlabs.github.io/sionna/_modules/sionna/rt/antenna.html#tr38901_pattern).

**Reply by nourhan0106** [reply]
I only see that we can only adjust the theta and phi and the model number and this gives me a gain of 1 for the iso pattern. I am not sure how to define the gain of the sensitivity.

def iso_pattern(theta, phi, slant_angle=0.0, polarization_model=2, dtype=tf.complex64):

**Reply by fanyimeng0** [reply]
I am not sure whether there is a built-in function to directly define antenna gain, but you can create a custom pattern to achieve the desired function. You can modify c = tf.ones_like(theta, dtype=dtype) by multiplying it with a 4dB gain to meet your first requirement. Similarly, you can adjust sensitivity and gain by customizing the code accordingly.

I'm not sure if this is the most efficient method, but it should work. Maintainers may have some better solution. https://nvlabs.github.io/sionna/em_primer.html?highlight=antenna%20gain

![image](https://github.com/user-attachments/assets/be484179-60cd-4eb6-9c14-8aaf22db6114)

**Reply by jhoydis** [reply]
Hi @nourhan0106,

The issue is slightly more involved. You cannot simply define the antenna gain by multiplying an isotropic antenna pattern by some real number. This would only change the radiation efficiency, which cannot be larger than one.

A gain of 4 dBi implies that the antenna is directing its energy more efficiently in certain directions compared to an isotropic radiator. Thus, an isotropic antenna has always a gain of 0dB.

The antenna gain is defined implicitly through by the antenna pattern [here](https://nvlabs.github.io/sionna/em_primer.html#equation-g). An example of an antenna pattern with higher gain is [this](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.antenna.tr38901_pattern).
